### PR TITLE
Custom class names in ReactCSSTransitionGroup

### PR DESCRIPTION
**Pull request by alexpien**
resolving #2680. API for adding custom classnames follows the proposed API in the issue.

### CODE REVIEW

File: src/addons/transitions/ReactCSSTransitionGroup.js [hunk 1]
```diff
@@ -27,7 +27,19 @@ var ReactCSSTransitionGroup = React.createClass({
...
+        leave: React.PropTypes.string
+      }),
+      React.PropTypes.shape({
+        enter: React.PropTypes.string,
+        enterActive: React.PropTypes.string,
+        leave: React.PropTypes.string,
+        leaveActive: React.PropTypes.string
+      })
+    ]).isRequired,
     transitionAppear: React.PropTypes.bool,
```
Comment (zpao) [inline src/addons/transitions/ReactCSSTransitionGroup.js]: We should make sure we support `appear` in the object as well.

File: src/addons/transitions/ReactCSSTransitionGroupChild.js [hunk 2]
```diff
@@ -48,8 +48,8 @@ var ReactCSSTransitionGroupChild = React.createClass({
 
   transition: function(animationType, finishCallback) {
     var node = React.findDOMNode(this);
-    var className = this.props.name + '-' + animationType;
-    var activeClassName = className + '-active';
+    var className = this.props.name[animationType] || this.props.name + '-' + animationType;
+    var activeClassName = this.props.name[animationType + 'Active'] || className + '-active';
```
Comment (zpao) [inline src/addons/transitions/ReactCSSTransitionGroupChild.js]: We should perhaps write some tests to make sure these work :)

**Comment by zpao** [comment]
@alexpien - would love to see this finished if you have the time!

**Comment by alexpien** [comment]
@zpao yea sorry for the delay I'll try to get this done sometime next week

File: src/addons/transitions/ReactCSSTransitionGroupChild.js (hunk 2 above)
Comment (alexpien) [inline src/addons/transitions/ReactCSSTransitionGroupChild.js]: I can't seem to get tests to work :/ any suggestions on how I should approach it? There aren't currently any tests to make sure the classnames are correct.

Comment (mosch) [inline src/addons/transitions/ReactCSSTransitionGroupChild.js]: @zpao can you help on this? I would also love to help out on this if i can. Maybe with the test?

Comment (zpao) [inline src/addons/transitions/ReactCSSTransitionGroupChild.js]: Ah, sorry! I missed this comment. Let's skip the test for now. Just make sure `appear` works and this should be good to go.

Comment (alexpien) [inline src/addons/transitions/ReactCSSTransitionGroupChild.js]: sounds good, ill make sure appear works and put something up today

**Comment by Daniel15** [comment]
:+1: 

This looks pretty good, thanks for the pull request :)

**Comment by zpao** [comment]
Sorry for the delay. I merged locally & pushed to avoid another roundtrip where we drop the ball again while you rebase. Thanks for your patience!

**Pull request merged**